sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -166,25 +209,28 @@ def test_distance_metrics_dtype_consistency(metric_param_grid):
 
         D64 = dm64.pairwise(X64)
         D32 = dm32.pairwise(X32)
-        assert_allclose(D64, D32)
+
+        # The original rtol is 1e-4 in of presence of float32.
+        # We increase the tolerance to 1e-6 to be stricter.
+        rtol = 1e-6
+        assert_allclose(D64, D32, rtol=rtol)
```

Comment:
But why not leaving `rtol` to its default value ?
It's kind of lucky that it passes for 1e-6 but it would probably fail if we tweak the datasets a bit.

Reply:
Actually this will fail both most datasets: both `D64.dtype` and `D32.dtype` are `np.float64` so the default `rtol` will be 1e-7 we need to set rtol explicitly to a lower value such as 1e-4 (to be lax) or 1e-5/1e-6 to be stricter after running the tests several rimes with different random seeds to generate the module-level test data.

Reply:
Should be fixed in 5ba0fbeea2a5c12e8f888bc20b37a0403d0ab40b.